server/src/main/java/io/crate/planner/operators/Collect.java

```diff
@@ -334,11 +334,7 @@ public LogicalPlan replaceSources(List<LogicalPlan> sources) {
     public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep) {
         LinkedHashSet<Symbol> newOutputs = new LinkedHashSet<>();
         for (Symbol outputToKeep : outputsToKeep) {
-            SymbolVisitors.intersection(outputToKeep, outputs, needle -> {
-                int index = outputs.indexOf(needle);
-                assert index != -1 : "Consumer is called only when intersection is found";
-                newOutputs.add(outputs.get(index));
-            });
+            SymbolVisitors.intersection(outputToKeep, outputs, newOutputs::add);
```

Comment:
Since this is reverting https://github.com/crate/crate/pull/16125:

I originally tested it via 
   ```
 CREATE TABLE tbl (x INT);

      SELECT
              TRY_CAST(x AS BIGINT) AS val
       FROM
           tbl as t
       WHERE
        x < 123
       ORDER BY
           val ASC;
```
Could you please try it locally? 

We have a unit test in https://github.com/crate/crate/pull/16125 but just to be sure (as we already had a case when unit test didn't reveal a problem).